metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1205,6 +1211,31 @@ class MetalsLanguageServer(
           scribe.error(s"error processing $sourceUri", e)
       }
     }
+    // Remove unused toplevel symbols from cache
+    tables.jarSymbols.deleteNotUsedTopLevels(usedJars.toArray)
```

Comment:
Can we schedule this asynchronously with 2 second delay? Indexing is on a critical path blocking the first compilation request so we should try to delay non-critical computation. It's also fine if this operation fails, the next startup can retry.

Reply:
Change it do delete asynchronously with 2 second delay.